packages/plugin-workspace-tools/sources/commands/foreach.ts

```diff
@@ -93,12 +95,15 @@ export default (clipanion: any, pluginConfiguration: PluginConfiguration) => cli
       const workspaces: Array<Workspace> = [];
 
       for (const workspace of candidates) {
-        if (!workspace.manifest.scripts.has(command))
+        if (command === 'run' && rest.length > 0 && !workspace.manifest.scripts.has(rest[0]))
```

Comment:
Can you move that into something like this? In the future I'd like to use the clipanion parser instead of assuming that `rest[0]` is the script name (it could be an option). It will be slightly easier to refactor it if the script name is computed in a single place:

```ts
const scriptName = command === `run` && rest.length > 0
  ? rest[0]
  : null;

if (!workspace.manifest.scripts.has(scriptName))
  // ...
```